src/providers/postgres/raster/qgspostgresrasterprovider.cpp

```diff
@@ -0,0 +1,1600 @@
+    return false;
+  }
+
+  // We first try to collect raster information using raster_columns information
+  // unless:
+  // - it is a query layer (unsupported at the moment)
+  // - use estimated metadata is false
+  // - there is a WHERE condition
+  // If previous conditions are not met or the first method fail try to fetch information
+  // directly from the raster data. This can be very slow.
+
+  // utility to get data type from string, used in both branches
+  auto pixelTypeFromString = [ ]( const QString & t ) -> Qgis::DataType
+  {
+    /* Pixel types
+    1BB - 1-bit boolean
+    2BUI - 2-bit unsigned integer
+    4BUI - 4-bit unsigned integer
+    8BSI - 8-bit signed integer
+    8BUI - 8-bit unsigned integer
+    16BSI - 16-bit signed integer
+    16BUI - 16-bit unsigned integer
+    32BSI - 32-bit signed integer
+    32BUI - 32-bit unsigned integer
+    32BF - 32-bit float
+    64BF - 64-bit float
+    */
+    Qgis::DataType type { Qgis::DataType::UnknownDataType };
+    if ( t == QStringLiteral( "8BUI" ) )
+    {
+      type = Qgis::DataType::Byte;
+    }
+    else if ( t == QStringLiteral( "16BUI" ) )
+    {
+      type = Qgis::DataType::UInt16;
+    }
+    else if ( t == QStringLiteral( "16BSI" ) )
+    {
+      type = Qgis::DataType::Int16;
+    }
+    else if ( t == QStringLiteral( "32BSI" ) )
+    {
+      type = Qgis::DataType::Int32;
+    }
+    else if ( t == QStringLiteral( "32BUI" ) )
+    {
+      type = Qgis::DataType::UInt32;
+    }
+    else if ( t == QStringLiteral( "32BF" ) )
+    {
+      type = Qgis::DataType::Float32;
+    }
+    else if ( t == QStringLiteral( "64BF" ) )
+    {
+      type = Qgis::DataType::Float64;
+    }
+    return type;
+  };
+
+  // ///////////////////////////////////////////////////////////////////
+  // First method: get information from metadata
+  if ( ! mIsQuery && mUseEstimatedMetadata && mSqlWhereClause.isEmpty() )
+  {
+    try
+    {
+      const QString sql { QStringLiteral( "SELECT r_raster_column, srid,"
+                                          "num_bands, pixel_types, nodata_values, ST_AsBinary(extent), blocksize_x, blocksize_y,"
+                                          "out_db, spatial_index, scale_x, scale_y, same_alignment,"
+                                          "regular_blocking "
+                                          "FROM raster_columns WHERE "
+                                          "r_table_name = %1 AND r_table_schema = %2" )
+                          .arg( quotedValue( mTableName ) )
+                          .arg( quotedValue( mSchemaName ) )  };
+
+      QgsPostgresResult result( connectionRO()->PQexec( sql ) );
+
+      if ( ( PGRES_TUPLES_OK == result.PQresultStatus() ) && ( result.PQntuples() > 0 ) )
+      {
+        mRasterColumn = result.PQgetvalue( 0, 0 );
+        mHasSpatialIndex = result.PQgetvalue( 0, 9 ) == 't';
+        bool ok;
+
+        mCrs = QgsCoordinateReferenceSystem::fromEpsgId( result.PQgetvalue( 0, 1 ).toLong( &ok ) );
```

Comment:
I think this should be createFromSrid. But keep in mind that postgis crs handling is rather broken in QGIS and needs to be rethought for 4.0 (we should be retrieving the definition from the server corresponding to the reported postgis srs, and not using any qgis internal databases for this ... which may or may not have any match to the server's definitions.)